tests/components/shopping_list/test_config_flow.py

```diff
@@ -0,0 +1,49 @@
+"""Test config flow."""
+
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_manual_config_set(hass):
+    """Test we ignore entry if manual config available."""
+    assert await async_setup_component(hass, "shopping_list", {})
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.flow.async_init(
+        "shopping_list", context={"source": "user"}
+    )
+    assert result["type"] == "abort"
```

Comment:
use const for the domain, "user" and "abort" and "form"

Reply:
There are no consts available for that in `homeassistant/const.py` and I can't find an example integration where these are imported?

Reply:
Look at iCloud/test_config_flow.py

Reply:
iCloud/test_config_flow.py is not using consts for "user" and "abort" and "form".

I did include the domain throgh my const.py though

Reply:
Nope, on dev :

user
`context={"source": SOURCE_USER}`

import
`context={"source": SOURCE_IMPORT}`

create_entry
`assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY`

form
`assert result["type"] == data_entry_flow.RESULT_TYPE_FORM`

abort
`assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT`

Reply:
Got it now. Thought you ment the ["type"] and so on